### ISSUE TITLE
Suggestions

### ISSUE THREAD

**Issue by sobhanattar** [santigarcor/laratrust#170]
I just wanted to **thank you** for this package. It's really good as far as I tested it, and that's not just for some additional features like **team** but because you are continually improving i. I think this a better choice than **Zizaco** which is left from 6 months ago and 313 issues!. And just for telling you how much I like this, I suggest two features you might like them:

- Encrypt permissions for roles and users so it can't be seen directly in database who has what permission.
- add **permission groups** entity for those bigger systems that wants to categorize their permissions for better UX. I already implementing this one in my own project and if you just let me know how I can merge it into your code, I'll be doing it.

Anyway thanks for this library, and I hope you continue to improve it 🥇

**Comment by KKSzymanowski** [comment]
- What would you want to achieve by encrypting permissions? Do you have some database users who don't have access to the codebase and cannot see the permissions?
- How are **permission groups** different from **roles**?

**Comment by santigarcor** [comment]
I'm with @KKSzymanowski, and what are the benefits for encrypting the roles permissions for everyone using this package?

**Comment by roelgonzalez94** [comment]
+1 for permission groups
I think what he believes is to have the option to couple some permissions into a group. For example if you have a specific section of students in a system. And you can assign a role with a the permission group of student-permissions which would be for example:

1. access students
2. remove students
3. update students
4. create students
5. add students to class
6. grade students

The role for example may be a teacher, but a teacher can do other things (permissions as well) and it would be beneficial when creating roles to just attach permission group to a role instead of individually assigning permissions.

I guess it will take a major rework in how the things work now.

**Comment by santigarcor** [comment]
I don't think it has to be a big rework, because we can leave the roles permissions attachment how it's now and add the groups permissions to it (That way the people can choose which one to use). Obviously the permissions check has to be updated to support that feature.

**Comment by roelgonzalez94** [comment]
Oh nice, yes I didn't think of such an easy solution! That works just fine.

**Comment by sobhanattar** [comment]
@roelgonzalez94 thanks for your explanation of what i was thinking. @santigarcor beside the above explanation, there is also another benefit for permission groups and that is categorizing related access levels for showing to user.

For example right now I'm working on a solution which have more than 300 permissions. What will happen to user if I show all these 300 permissions! instead I added some permission_groups table to laratrust, and put each permission in on permission group. Now when I want to show an admin I can show him group based permissions like the one @roelgonzalez94 said:

```
**Post**
- Create Post
- Update Post
- ...
**News**
- Create News
- Update News
- ...
```
This way my End user will have easier time to go through permissions.

As of encrypting the permissions, the scenario can be like what @KKSzymanowski said - some other user of database - and also the case - which might be a false idea of security - is the case that our database is bridged. 

As far as I see the only place that we can encrypt our data is in cache and maybe one or two relational tables that some can look into and see, who has what permissions and start logging by their usernames.

**Comment by deivisonrpg** [comment]
+1 for permission groups is a good addiction

**Comment by zeraist** [comment]
+1 for permission groups

**Comment by leandroruel** [comment]
update to Laravel 5.5 too...

**Comment by santigarcor** [comment]
@leandroruel you can use it in laravel 5.5 and in your custom middleware

**Comment by reanzi** [comment]
+1 for permission groups
